### PR TITLE
Added strict versons for llvm-hs-* dependencies

### PR DESCRIPTION
**Pull request by LightAndLight**
Grin currently only builds using llvm-hs 5.*, so we tell cabal that in the .cabal file.

### CODE REVIEW

**Comment by LightAndLight** [comment]
I'll resubmit this pull request from a different branch

File: grin/grin.cabal [hunk 1]
```diff
@@ -130,8 +130,9 @@ library
     monad-gen,
     free,
     bytestring,
-    llvm-hs,
-    llvm-hs-pure,
+    llvm-hs ==5.*,
```
Comment (csabahruska) [inline grin/grin.cabal]: GRIN uses llvm 7
https://www.stackage.org/nightly-2018-11-07/package/llvm-hs-7.0.1
https://github.com/grin-tech/grin/blob/master/stack.yaml#L1

Comment (csabahruska) [reply grin/grin.cabal]: Do not specify the version in the cabal file.
Stack fixes the versions.

Comment (LightAndLight) [reply grin/grin.cabal]: Okay thanks, I'll fix things up accordingly.

Comment (LightAndLight) [reply grin/grin.cabal]: @csabahruska I don't use stack to build my code. Do you really prefer I leave the versions out of the cabal file?

Comment (csabahruska) [reply grin/grin.cabal]: Can't you set the version in the nix file?

Comment (LightAndLight) [reply grin/grin.cabal]: Nix works without putting the version in .cabal. It's just by having the version in the cabal file, cabal will tell me if I nix uses the wrong version by mistake.

Comment (LightAndLight) [reply grin/grin.cabal]: Also, I'd suggest putting a lower bound on `base` because this codebase doesn't build on GHC8.2

**Pull request closed without merging**